### PR TITLE
Distributed error reporting: Create model to store captured Errors

### LINKED ISSUE #12257: Implement the ErrorReport model
## Overview
This issue implements a model that stores error reports

## Description and outcomes
- Create  a new model named `ErrorReports` in the `errorreports` app with the following attributes:
    - `error_from`: To know if the error is from frontend or the backend
        - Type: `CharField`, `Enum`
        - Length: 10, either frontend or backend
    - `error_message`: The actual error message, like ‘Operational Error’
        - Type: `CharField`
        - Length: 255
    - `traceback`, containing meaningful stack traces
        - `Type`: `TextField`
    - `first_occured`: The time the error first occurred in UTC
        - Type: `DateTimeField`
        - Default: `timezone.now`
    - `last_occured`: If the same error reoccurs, this will be set to the time of recurrence, otherwise the `first_occured` time is set
        - Type: `DateTimeField`
        - Default: `timezone.now`
    - `sent`: Send the status of the error.
        - Type: `BooleanField`
        - Default: `False`
    - `no_of_errors`: Number of occurrences of a specific error. This will help with the prioritization of errors
        - Type: `IntegerField`
        - Default: 1.
- Run the migrations for the model
- Create the following methods to manipulate the `ErrorReports`. These methods shall be `classmethods`, defined inside `ErrorReports` class.
    - `insert_or_update_error`: This shall create a new error object, and also update an existing error if `error_message` and `traceback` match with an existing object. Even if `sent` is `true`, update `last_occured` and `no_of_errors`
    - `get_unsent_errors`: This shall return the tuple of errors that have `sent` field as `false`(unreported errors)
    - `delete_error`: A little out-of-scope for now, as the deletion logic of reported errors is not yet finalized(Might be implemented with cronJob or something similar later on)
    - `mark_as_sent`: It should set the sent field of the object. So need not be a class method
- For each of the methods above, If `DEVELOPER_MODE` is `true`, then skip updating the model and just log the error else update the model

## Acceptance Criteria
- Model is migrated
- `insert_or_update_error()' and `get_unsent_errors()` are implemented as mentioned. `delete_error()` can just do nothing for now(add ellipsis)
- Test are written to verify correctness of the created model, ensure that the methods below work correctly and that errors are only logged into the database in `DEVELOPER_MODE`. The methods include;
    - `insert_or_update_error()`
    - `get_unsent_errors()`
    - `mark_as_sent()`
 
## Assumptions and Dependencies
- Django will handle the table updating

## Scope
- Creating new model
- Running migration
- Defining the `db_utils` functions
- Writing tests to ensure the functions work properly

## Accessibility Requirements
NA

## Resources
- [Django Models](https://docs.djangoproject.com/en/3.2/topics/db/models/)
- [Class Methods](https://docs.djangoproject.com/en/3.2/topics/db/models/)

### PR DESCRIPTION
**Pull request by thesujai**
## Summary

1. Adds new model `ErrorReports` for storing reported errors
2. Add Class methods(utils) for manipulating ErrorReports
3. New tests for the model methods 
**Screenshots**
Test passed:
Screenshot from 2024-06-07 15-56-33
Functionality Passed:
Screenshot from 2024-06-07 15-56-16

## References

Closes #12257 

## Reviewer guidance

…

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by thesujai** [comment]
@akolson @rtibbles Ready for review!

File: kolibri/core/errorreports/models.py [hunk 1]
```diff
@@ -36,3 +45,50 @@ def allow_migrate(self, db, app_label, model_name=None, **hints):
...
+        logger.error("ErrorReports: Database not updated, as DEVELOPER_MODE is True.")
+        return None
+
+    @classmethod
+    def get_unsent_errors(cls):
+        return cls.objects.filter(sent=False)
+
+    @classmethod
+    def delete_error(cls):
+        ...
```
Comment (akolson) [inline kolibri/core/errorreports/models.py]: Lets use `pass` here!

**Review by akolson — CHANGES_REQUESTED** [review]
Generally PR looks good to me! Just one blocking change and I think we should be good for the merge! Thanks @thesujai !

File: kolibri/core/errorreports/models.py (hunk 1 above)
Comment (thesujai) [reply kolibri/core/errorreports/models.py]: ok. I ain't using ellipsis ever again

File: kolibri/core/errorreports/models.py [hunk 2]
```diff
@@ -36,3 +45,50 @@ def allow_migrate(self, db, app_label, model_name=None, **hints):
...
+    def __str__(self):
+        return f"{self.error_message} ({self.error_from})"
+
+    def mark_as_sent(self):
+        self.sent = True
+        self.save()
+
+    @classmethod
+    def insert_or_update_error(cls, error_from, error_message, traceback):
+        if not settings.DEVELOPER_MODE:
```
Comment (akolson) [inline kolibri/core/errorreports/models.py]: Can we use `getattr(settings, "DEVELOPER_MODE", None)` instead based the issues we discussed?

Comment (thesujai) [reply kolibri/core/errorreports/models.py]: Done and tested as well

**Review by akolson — APPROVED** [review]
LGTM! Thanks @thesujai

**Pull request merged**